sources/models/src/modeled_types/shared.rs

```diff
@@ -1130,3 +1133,44 @@ mod test_valid_kmod_key {
         assert!(KmodKey::try_from(vec!["z"; KMOD_KEY_LENGTH + 1].join("")).is_err());
     }
 }
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// Input value that needs to be a positive value, but should not be greater
+/// than an i32::MAX.
+#[derive(Debug, PartialEq, Scalar)]
+pub struct PositiveInteger {
```

Comment:
Should this be named `NonNegativeInteger` since zero is allowed and is neither positive or negative?

I realize that's a bit of a math nitpick but I could see someone assuming that the PositiveInteger type would exclude zero without looking too closely at the implementation.

Reply:
Renamed to appease all mathematical purists. :D